File: go-client-sdk/pkg/models/operations/getcompaniescompanyiddatajournalsjournalid.go
package operations

import (
	"github.com/speakeasy-sdks/codat-sdks/go-client-sdk/v2/pkg/models/shared"
)

type GetCompaniesCompanyIDDataJournalsJournalIDPathParams struct {
	CompanyID string `pathParam:"style=simple,explode=false,name=companyId"`
	JournalID string `pathParam:"style=simple,explode=false,name=journalId"`
}

type GetCompaniesCompanyIDDataJournalsJournalIDSecurity struct {
	APIKeyAuth *shared.SchemeAPIKeyAuth `security:"scheme,type=apiKey,subtype=header"`
	CodatLogin *shared.SchemeCodatLogin `security:"scheme,type=oauth2"`
}

type GetCompaniesCompanyIDDataJournalsJournalIDRequest struct {
	PathParams GetCompaniesCompanyIDDataJournalsJournalIDPathParams
	Security   GetCompaniesCompanyIDDataJournalsJournalIDSecurity
}

type GetCompaniesCompanyIDDataJournalsJournalIDResponse struct {
	CodatDataContractsDatasetsJournalPagedResponseModel *shared.CodatDataContractsDatasetsJournalPagedResponseModel
	ContentType                                         string
	StatusCode                                          int
}

File: typescript-client-sdk/src/sdk/models/operations/getcompaniescompanyiddatajournalsjournalid.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class GetCompaniesCompanyIdDataJournalsJournalIdPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=companyId" })
  companyId: string;

  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=journalId" })
  journalId: string;
}


export class GetCompaniesCompanyIdDataJournalsJournalIdSecurity extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "security, scheme=true;type=oauth2" })
  codatLogin?: shared.SchemeCodatLogin;

  @SpeakeasyMetadata({ data: "security, scheme=true;type=apiKey;subtype=header" })
  apiKeyAuth?: shared.SchemeApiKeyAuth;
}


export class GetCompaniesCompanyIdDataJournalsJournalIdRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: GetCompaniesCompanyIdDataJournalsJournalIdPathParams;

  @SpeakeasyMetadata()
  security: GetCompaniesCompanyIdDataJournalsJournalIdSecurity;
}


export class GetCompaniesCompanyIdDataJournalsJournalIdResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  codatDataContractsDatasetsJournalPagedResponseModel?: shared.CodatDataContractsDatasetsJournalPagedResponseModel;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;
}
